test/src/test/java/hudson/model/AbstractItemTest.java

```diff
@@ -132,6 +138,30 @@ public void renameViaRestApi() throws Exception {
         assertThat(p.getName(), equalTo("bar"));
     }
 
+    @Test
+    @Issue("JENKINS-52164")
+    public void renameLinksShouldBeValid() throws Exception {
+        FreeStyleProject project1 = j.createFreeStyleProject("project1");
+        MockFolder folder1 = j.createProject(RenameableMockFolder.class, "folder1");
+        FreeStyleProject project2 = folder1.createProject(FreeStyleProject.class, "project2");
+
+        HtmlAnchor anchor = findRenameAnchor(project1);
+        anchor.click();
+
+        anchor = findRenameAnchor(project2);
+        anchor.click();
+
+        anchor = findRenameAnchor(folder1); // Throws ElementNotFoundException before JENKINS-52164 fix
```

Comment:
MockFolder has no UI, so this is still failing. Need to come up with a better way to test it, or just move the test to cloudbees-folder and use a SNAPSHOT to validate the core change.

Reply:
What about something like this?

```
Action renameAction = null;
        for (Action action : Jenkins.get().getActions()) {
            if (action instanceof RenameAction) {
                renameAction = action;
                break;
            }
        }

        if (renameAction != null) {
            assertTrue(Functions.getActionUrl(folder1.getUrl(), renameAction).equals(j.contextPath + "/" + folder1.getUrl() + folder1.getAction(RenameAction.class).getUrlName()));
        }
```

It would be checking the URL generated in the jelly file with the one simulated in the in method `findRenameAnchor`

Reply:
My only issue with that approach is that that test would pass before or after the change to `RenameAction/action.jelly`. I will probably just move these tests to cloudbees-folder.

Reply:
You're probably right and if we want to test the UI, you'll have to move the test to the plugin